src/main/java/com/epam/service/TraineeService.java

```diff
@@ -3,12 +3,18 @@
 import com.epam.model.Trainee;
 
 import java.util.List;
+import java.util.Optional;
 
 public interface TraineeService {
 
-    void createTrainee(Trainee trainee);
-    void updateTrainee(Trainee trainee);
-    void deleteTrainee(Integer traineeId);
-    Trainee getTraineeById(Integer traineeId);
+    Trainee createTrainee(Trainee trainee);
+    Optional<Trainee> updateTrainee(Trainee trainee);
+    Optional<Trainee> getTraineeById(Integer traineeId);
+    Optional<Trainee> getTraineeByUsername(String username);
     List<Trainee> getAllTrainees();
+    boolean deleteTrainee(Integer traineeId);
+    boolean deleteTraineeByUsername(String username);
+    boolean changePassword(Integer traineeId, String currentPassword, String newPassword);
+    void activateDeactivateTrainee(Integer traineeId, boolean activate);
```

Comment:
I would recommend to have different methods `activate` and `deactivate`. The `activateDeactivateTrainee(1, true)` call could have ambiguous meaning for other developers. And the will need to take some time and assert that such a call will activate a trainee and not deactivate it. Having to different methods each one for certain action removes this ambiguity. Please refactor this for all services.